src/com/ConnectionInfoPublisher.cpp

```diff
@@ -49,56 +50,103 @@ std::string ConnectionInfoPublisher::getLocalDirectory() const
 
 std::string ConnectionInfoPublisher::getFilename() const
 {
-  using namespace boost::filesystem;
-
   auto local  = getLocalDirectory();
   auto hashed = impl::hashedFilePath(acceptorName, requesterName, tag, rank);
-  path p      = path(getLocalDirectory()) / path(hashed);
+  auto p      = bfs::path(getLocalDirectory()) / hashed;
 
   return p.string();
 }
 
 std::string ConnectionInfoReader::read() const
 {
-  std::ifstream ifs;
-  auto          path = getFilename();
-  PRECICE_DEBUG("Waiting for connection file {}", path);
-  do {
-    ifs.open(path, std::ifstream::in);
-    std::this_thread::sleep_for(std::chrono::milliseconds(1));
-  } while (not ifs);
-  PRECICE_DEBUG("Found connection file {}", path);
+  auto path = getFilename();
 
+  PRECICE_DEBUG("Waiting for connection file \"{}\"", path);
+  const auto waitdelay = std::chrono::milliseconds(1);
+  while (!bfs::exists(path)) {
+    std::this_thread::sleep_for(waitdelay);
+  }
+  PRECICE_ASSERT(bfs::exists(path));
+  PRECICE_DEBUG("Found connection file \"{}\"", path);
+
+  std::ifstream ifs(path);
+  PRECICE_CHECK(ifs,
+                "Unable to establish connection as the connection file \"{}\" couldn't be opened.",
+                path);
   std::string addressData;
-  ifs >> addressData;
+  std::getline(ifs, addressData);
+  PRECICE_CHECK(!addressData.empty(),
+                "Unable to establish connection as the connection file \"{}\" is empty. "
+                "Please report this bug to the preCICE developers.",
+                path);
+  boost::algorithm::trim_right(addressData);
   return addressData;
 }
 
 ConnectionInfoWriter::~ConnectionInfoWriter()
 {
-  namespace fs = boost::filesystem;
+  bfs::path path(getFilename());
+  if (!bfs::exists(path)) {
+    PRECICE_WARN("Cannot clean-up the connection file \"{}\" as it doesn't exist. "
+                 "In case of connection problems, please report this to the preCICE developers.",
+                 path);
+    return;
+  }
+  PRECICE_DEBUG("Deleting connection file \"{}\"", path);
   try {
-    fs::path p(getFilename());
-    PRECICE_DEBUG("Deleting connection file {}", p.string());
-    fs::remove(p);
-  } catch (const fs::filesystem_error &e) {
-    PRECICE_WARN("Unable to delete connection file due to error: {}", e.what());
+    bfs::remove(path);
+    /// @TODO Check if we need to sleep to let the underlying filesystem synchronize
```

Comment:
Do you want to check this before merging? How do you want to check this?

Reply:
We need to run a test case with a distributed filesystem running on separated nodes preferably with a slow inter-node connection.

Reply:
For the record https://github.com/precice/precice/pull/1290#pullrequestreview-1060192506